core/tests/protractor_utils/AdminPage.js

```diff
@@ -268,6 +306,139 @@ var AdminPage = function() {
       expect(name).toEqual(expectedUsernamesArray[j]);
     }
   };
+
+  this.getMiscTab = async function() {
+    await miscTabButton.click();
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.uploadTopicSimilarities =
+    async function(relativePathToSimilaritiesFile) {
+      absPath = path.resolve(__dirname, relativePathToSimilaritiesFile);
+      await waitFor.visibilityOf(chooseSimilarityFileInput,
+        'Similarity upload form is not visible');
+      await chooseSimilarityFileInput.sendKeys(absPath);
+      await waitFor.elementToBeClickable(similarityFileUploadButton,
+        'Upload button taking too long to be clickable');
+      await similarityFileUploadButton.click();
+    };
+
+  this.expectSimilaritiesToBeUploaded = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Topic similarities uploaded successfully.');
+    return true;
+  };
+
+  this.expectUploadError = async function() {
+    await statusMessage.getText().then((text) => {
+      if (text.match('Server')) {
+        return true;
+      }
+      return false;
+    });
+  };
+
+  this.downloadSimilarityFile = async function() {
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Download similarity file button not clickable');
+    await similarityDownloadButton.click();
+  };
+
+  this.expectFileToBeDownloaded = async function() {
+    var filename = global.downloadsPath + '/topic_similarities.csv';
+    browser.driver.wait(function() {
+      return fs.existsSync(filename);
+    }, 30000);
```

Comment:
Could you make this a general function in `waitFor` in case it's useful for other people?

Reply:
done!